core/src/main/java/com/linecorp/armeria/server/HttpStatusException.java

```diff
@@ -32,51 +34,69 @@
 
     static {
         for (int i = 0; i < EXCEPTIONS.length; i++) {
-            EXCEPTIONS[i] = new HttpStatusException(HttpStatus.valueOf(i), false);
+            EXCEPTIONS[i] = new HttpStatusException(HttpStatus.valueOf(i), false, null);
         }
     }
 
     /**
      * Returns a new {@link HttpStatusException} instance with the specified HTTP status code.
      */
     public static HttpStatusException of(int statusCode) {
-        if (statusCode < 0 || statusCode >= 1000) {
-            final HttpStatus status = HttpStatus.valueOf(statusCode);
-            if (Flags.verboseExceptionSampler().isSampled(HttpStatusException.class)) {
-                return new HttpStatusException(status);
+        return of(HttpStatus.valueOf(statusCode));
+    }
+
+    /**
+     * Returns a new {@link HttpStatusException} instance with the specified HTTP status code and {@code cause}.
+     */
+    public static HttpStatusException of(int statusCode, Throwable cause) {
```

Comment:
```suggestion
    public static HttpStatusException of(int statusCode, Throwable cause) {
        requireNonNull(cause, "cause");
```

I think normally we check null at all public APIs even if a delegate would check it.